src/get_v2_vs_frac.py

```diff
@@ -0,0 +1,298 @@
+"""
+python script for the computation of the prompt or non-prompt v2 via extrapolation
+run: python ComputeV2vsFDFrac.py config.yaml --inputdir path/to/input --outputdir path/to/output --suffix text
+
+"""
+import argparse
+import os
+import yaml
+import sys
+from ROOT import TFile, TCanvas, TLegend, TLatex, TGraphErrors, TF1, TH1D, TVirtualFitter, Double_t, gROOT
+from ROOT import kBlack, kAzure, kOrange
+from ROOT import kFullCircle
+script_dir = os.path.dirname(os.path.abspath(__file__))
+sys.path.append(os.path.join(script_dir, '..', 'utils'))
+from utils import get_particle_info
+from StyleFormatter import SetObjectStyle, GetROOTColor
+
+def load_ry_files(inputdir):
+    if os.path.exists(f'{inputdir}'):
+        print(f'Loading v2 files from {inputdir}')
+        v2Files = [f'{inputdir}/{file}'
+                    for file in os.listdir(f'{inputdir}') if file.endswith('.root')]
+        print(os.listdir(f'{inputdir}'))
+        v2Files.sort()
+    else:
+        raise ValueError(f'No raw_yields folder found in {inputdir}')
+    return v2Files
+
+def load_frac_files(inputdir):
+    if os.path.exists(f'{inputdir}'):
+        print(f'Loading frac files from {inputdir}')
+        fracFiles = [f'{inputdir}/{file}'
+                        for file in os.listdir(f'{inputdir}') if file.endswith('.root')]
+        fracFiles.sort()
+    else:
+        raise ValueError(f'No frac folder found in {inputdir}')
+    return fracFiles
+
+def set_frame_style(canv, Title, particleTit):
+    canv.SetLeftMargin(0.15)
+    canv.SetRightMargin(0.05)
+    canv.SetBottomMargin(0.15)
+    canv.SetTopMargin(0.05)
+    hFrame = canv.DrawFrame(0.0, -0.2, 1, 0.35, f";Non-prompt {particleTit} fraction; #it{{v}}_{{2}}^{{#it{{obs}}}}")
+    hFrame.GetYaxis().SetDecimals()
+    hFrame.GetYaxis().SetNoExponent()
+    hFrame.GetXaxis().SetMoreLogLabels()
+    hFrame.GetYaxis().SetTitleSize(0.04)
+    hFrame.GetYaxis().SetTitleOffset(1.2)
+    hFrame.GetYaxis().SetLabelSize(0.04)
+    hFrame.GetXaxis().SetTitleSize(0.04)
+    hFrame.GetXaxis().SetLabelSize(0.04)
+    hFrame.GetXaxis().SetTitleOffset(1.2)
+    hFrame.GetYaxis().SetNdivisions(505)
+
+def set_frame_margin(canv):
+    canv.SetLeftMargin(0.15)
+    canv.SetRightMargin(0.05)
+    canv.SetBottomMargin(0.15)
+    canv.SetTopMargin(0.05)
+
+def v2_vs_frac(flow_config, rawYieldFiles, fracFiles, correlated, batch=False):
+
+    gROOT.SetBatch(True)
+    
+    #!!!!!!!! totall drop the uncorrelated cutsets change /home/wuct/ALICE/local/dev/hf-vn/src/cut_variation.py
+    # load the configuration
+    ptmins, ptmaxs, nPtBins, particleName, particleTit, decay, CutSets = None, None, None, None, None, None, None
+    with open(flow_config, 'r') as f:
+        cfg = yaml.safe_load(f)
+        
+        ptmins = cfg['ptbins'][:-1]
+        ptmaxs = cfg['ptbins'][1:]
+        nPtBins = len(ptmins)
+        particleName = cfg['Dmeson']
+        particleTit, _, decay, _, _, _ = get_particle_info(particleName)
```

Comment:
```suggestion
        particleTit, _, decay, _, _, _ = get_particle_info(cfg['Dmeson'])
```

so we can also avoid initializing `particleName = None` above